clang/test/CIR/CodeGen/no-prototype.c

```diff
@@ -0,0 +1,51 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir-enable -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
```

Comment:
Very nice testcase, thanks for breaking into several possible cases.

Can you double check that all C standard versions support no-prototype idioms? If not we need the tests here to catch the right ones.

Reply:
@bcardosolopes looking at the [docs](https://clang.llvm.org/c_status.html) and testing it, it seems that it works up to C17/C18, then on C2X it is handled like C++ (`int func()` defaults to `int func(void)`).

I'll add a test case for `-std=c2x` too.